operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -0,0 +1,342 @@
+import com.ibm.fhir.model.resource.Patient;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.type.UnsignedInt;
+import com.ibm.fhir.model.type.Uri;
+import com.ibm.fhir.model.type.code.BundleType;
+import com.ibm.fhir.model.type.code.IssueType;
+import com.ibm.fhir.search.compartment.CompartmentUtil;
+import com.ibm.fhir.search.exception.FHIRSearchException;
+import com.ibm.fhir.server.operation.spi.AbstractOperation;
+import com.ibm.fhir.server.operation.spi.FHIROperationContext;
+import com.ibm.fhir.server.operation.spi.FHIRResourceHelpers;
+import com.ibm.fhir.server.util.FHIROperationUtil;
+
+
+/**
+ * This class implements the <a href="https://www.hl7.org/fhir/operation-patient-everything.html">$everything</a> operation
+ * which is used to return all the information related to one or more patients described in the resource or context on 
+ * which this operation is invoked.
+ * 
+ * @author Luis A. García
+ */
+public class EverythingOperation extends AbstractOperation {
+
+    private static final Logger LOG = java.util.logging.Logger.getLogger(EverythingOperation.class.getName());
+    
+    /**
+     * The <a href="https://www.hl7.org/fhir/search.html#prefix">prefix</a> used to indicate the start date for the $everything resources
+     */
+    protected static final String STARTING_FROM = "ge";
+
+    /**
+     * The <a href="https://www.hl7.org/fhir/search.html#prefix">prefix</a> used to indicate the end date for the $everything resources
+     */
+    protected static final String UP_UNTIL = "le";
+
+    /**
+     * The "date" query parameter used in the underlying search operation.
+     */
+    protected static final String DATE_QUERY_PARAMETER = "date";
+
+    /**
+     * The "_lastUpdated" query parameter used in the underlying search operation.
+     */
+    protected static final String LAST_UPDATED_QUERY_PARAMETER = "_lastUpdated";
+
+    /**
+     * The query parameter to indicate a maximum resource count for the $everything operation
+     */
+    protected static final String COUNT_QUERY_PARAMETER = "_count";
+
+    /**
+     * The query parameter to indicate the resource types for the $everything operation
+     */
+    protected static final String TYPE_QUERY_PARAMETER = "_type";
+
+    /**
+     * The query parameter to indicate a start date for the $everything operation
+     */
+    protected static final String START_QUERY_PARAMETER = "start";
+
+    /**
+     * The query parameter to indicate a stop date for the $everything operation
+     */
+    protected static final String END_QUERY_PARAMETER = "end";
+
+    /**
+     * The query parameter to only return resources last update since a date for the $everything operation
+     */
+    protected static final String SINCE_QUERY_PARAMETER = "_since";
+
+    /**
+     * The default max number of resources to search for each subsresource
+     */
+    protected static final int DEFAULT_RESOURCE_COUNT = 500;
```

Comment:
This should be maximum 1000 and come from the SearchConstants class

Reply:
Done.